src/resource/worker-control.c

```diff
@@ -643,17 +642,6 @@ sid_resource_t *worker_control_get_new_worker(sid_resource_t *worker_control_res
 		goto out;
 	}
 
-	if (sigfillset(&new_sigmask) < 0) {
-		log_sys_error(ID(worker_control_res), "sigfillset", "");
-		goto out;
-	}
-
-	if (sigprocmask(SIG_SETMASK, &new_sigmask, &original_sigmask) < 0) {
-		log_sys_error(ID(worker_control_res), "sigprocmask", "blocking signals before fork");
-		goto out;
-	}
-	signals_blocked = 1;
```

Comment:
But here we were blocking all possible signals. The other parts of the code block only certain subset of the whole signal set.

The `worker-control` should encapsulate everything that is needed for the workers to get created and managed without any further dependencies and intervention from outside, so the outside code simply doesn't need to bother. IOW, the code inside `worker-control` shouldn't probably assume that the "outside" has done everything that `worker-control` needs for its proper functionality.

Reply:
I see.  I'll drop these changes.